code/graph-algorithms/diameter_trees/cpp/Diameter.cpp

```diff
@@ -0,0 +1,65 @@
+//Learn O(n)
+
+
+// The following is O(n^2)
+#include<bits/stdc++.h>
+#define ll long long
+using namespace std;
+
+typedef struct tree node;
+
+struct tree
+{
+	ll data;
+	node * right;
+	node * left;
+};
+
+node * insert(ll a)
```

Comment:
This function name seems a bit misleading as it actually creates a new node rather than inserting it in something.